R/crew_launcher_lsf.R

```diff
@@ -0,0 +1,287 @@
+#' @title `r lifecycle::badge('experimental')` Create a launcher with
+#'   LSF workers.
+#' @export
+#' @family launchers
+#' @description Create an `R6` object to launch and maintain
+#'   workers as LSF jobs.
+#' @details WARNING: the `crew.cluster` LSF plugin is experimental
+#'   and has not actually been tested on a LSF cluster. Please proceed
+#'   with caution and report bugs to
+#'   <https://github.com/wlandau/crew.cluster>.
+#'
+#'   To launch a LSF worker, this launcher
+#'   creates a temporary job script with a call to `crew::crew_worker()`
+#'   and submits it as an LSF job with `sbatch`. To see most of the lines
+#'   of the job script in advance, use the `script()` method of the launcher.
+#'   It has all the lines except for the job name and the
+#'   call to `crew::crew_worker()`, both of
+#'   which will be inserted at the last minute when it is time
+#'   to actually launch a worker.
+#' @inheritParams crew_launcher_cluster
+#' @param lsf_log_output Character of length 1, file pattern to control
+#'   the locations of the LSF worker log files. By default, both standard
+#'   output and standard error go to the same file.
+#'   `lsf_log_output = "crew_log_%J.log"` translates to a line of
+#'   `#BSUB-o crew_log_%J.log` in the LSF job script,
+#'   where `%J` is replaced by the job ID of the worker.
+#'   The default is `/dev/null` to omit these logs.
+#'   Set `lsf_log_output = NULL` to omit this line from the job script.
+#' @param lsf_log_error Character of length 1, file pattern for standard
+#'   error. `lsf_log_error = "crew_error_%J.err"` translates to a line of
+#'   `#BSUB-e crew_error_%J.err` in the LSF job script,
+#'   where `%J` is replaced by the job ID of the worker.
+#'   The default is `/dev/null` to omit these logs.
+#'   Set `lsf_log_error = NULL` to omit this line from the job script.
+#' @param lsf_memory_megabytes_per_cpu Positive numeric of length 1
```

Comment:
I see you carried this over from SLURM, and I think we can call it something like `lsf_memory_megabytes` unless it is really per CPU (is it really megabytes?)

Reply:
On my cluster the default is in megabytes, but it looks like this varies by environment. It looks like providing units explicitly is an option: https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=o-m

Reply:
An on reflection, it would also be nice to have arguments `lsf_memory_gigabytes_required` and `lsf_cores` which are both positive numerics and which translate into the appropriate lines with the `-R` flag.